syne_tune/optimizer/schedulers/searchers/searcher.py

```diff
@@ -577,11 +599,43 @@ def get_batch_configs(self, batch_size: int, **kwargs):
                 configs.append(config)
         return configs
 
-    def _validate_config_space(self, config_space: dict):
-        # GridSearcher only supports Categorical hyperparameters for now
-        for hp_range in config_space.values():
+    def _validate_config_space(self, config_space: dict, num_samples: dict):
+        '''
+        This function validates config_space from two aspects: first, that all hyperparameters
+        are of acceptable types (i.e. float, integer, categorical). Second, num_samples is specified
+        for float hyperparameters. Num_samples for categorical variables are ignored as all of their
+        values is used in GridSearch. Num_samples for integer variables is optional, if specified it
+        will be used and will be capped at their range length.
+        Args:
+            config_space: The configuration space that defines search space grid.
+            num_samples: Number of samples for each hyperparmaters.Only required for float hyperparameters.
+        '''
+        if num_samples is None:
+            num_samples = {}
+        self.num_samples = num_samples
+        for hp, hp_range in config_space.items():
+            # Acceptable types for hyperparameters are Float, Integer, Categorical.
```

Comment:
Aren't we missing FiniteRange here? I'd treat it the same as Categorical: Use all values